plugin/Model/FormModelBuilder/FormModelBuilder.php

```diff
@@ -123,19 +129,35 @@ protected function readTemplatePaths($directory, $pattern = '*')
 			.'/onoffice-theme/templates/'.$directory.'/'.$pattern.'.php');
 		
 		$templatesAll = array_merge($templateGlobFiles, $templateLocalFiles, $templateThemeFiles);
-		$templates = array();
-
-		foreach ($templatesAll as $value) {
-			if(strpos($value, 'themes') !== false) {
-				$value = __String::getNew($value)->replace(get_template_directory().'/', '');
-				$templates[$value] = $value;
-			}else{
-				$value = __String::getNew($value)->replace(plugin_dir_path(ONOFFICE_PLUGIN_DIR), '');
-				$templates[$value] = $value;
+		$templateFolderData = array();
```

Comment:
This is a very central place to put this rather special code. It'd be best if it was somewhere where it actually belongs and if it was tested.